iphone/Maps/Core/TextToSpeech/MWMTextToSpeech.mm

```diff
@@ -279,6 +282,43 @@ - (AVAudioPlayer *)audioPlayer {
   return _audioPlayer;
 }
 
+- (void)playRandomTestString {
+  if (![self isValid])
+    [self createVoice:[[self class] savedLanguage]];
+  
+  if (self.testStrings == nil)
+    return;
+  
+  [self speakOneString: self.testStrings[_testStringIndex]];
+  
+  _testStringIndex++;
+  if (_testStringIndex >= self.testStrings.count)
+    _testStringIndex = 0;
+}
+
+- (NSArray *)testStrings {
+  NSString * language = [NSString stringWithUTF8String: bcp47ToTwineLanguage([[self class] savedLanguage]).c_str()];
+  NSString * languagePath = [NSBundle.mainBundle pathForResource: language ofType: @"lproj"];
+  if (languagePath == nil) {
+    LOG(LWARNING,
+        ([@"Couldn't find translation file for " stringByAppendingString: language]));
+    return nil;
+  }
+  NSBundle * bundle = [NSBundle bundleWithPath: languagePath];
```

Comment:
Is this trickery above really needed to load localized strings?

Reply:
I was also a bit surprised but I think that's the only way (see [here](https://stackoverflow.com/questions/27879232/force-nslocalizedstring-to-use-a-specific-language-using-swift))